### ISSUE TITLE
[Usability bug] Menu automatically gets closed after clearing the list of recent files.

### ISSUE THREAD

**Issue by Vahan86** [linuxmint/Cinnamon#4595]
When you open Menu and press "Clear list" entry in Menu -> "Recent files" sub-menu to clear the list of recent files, the Menu automatically gets closed after clearing the files.
It's a little bit inconvenient, because if you want to switch to other sub-menus (for instance Places, Preferences, etc.) you have to open Menu again.

It would be better if Menu stays open after clearing the list of files in "Recent files" sub-menu.

Thanks.

**Comment by Vahan86** [comment]
The bug still exists in Cinnamon 3.0.7

**Comment by NikoKrause** [comment]
I'm not sure if it's a bug or if it's a feature.
This could be done by deleting one line.

**Comment by Vahan86** [comment]
Checked in Cinnamon 3.2.6. The issue is still there.

**Comment by NikoKrause** [comment]
I know. This doesn't look to be difficult to solve at first glance. It's quite easy to leave the menu open, when clearing the list. The problem I ran into with this one though is, that the keynavigation breaks, when we don't close the menu after clearing the list. I think I know, which parameters need to be changed, but I didn't figure out the right values for the parameters, yet.

I'm not sure if this was the only reason, why it was implemented this way.
